datafusion/sql/src/expr/mod.rs

```diff
@@ -714,6 +717,37 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         not_impl_err!("Unsupported dictionary literal: {raw_expr:?}")
     }
 
+    fn try_plan_map_literal(
+        &self,
+        entries: Vec<MapEntry>,
+        schema: &DFSchema,
+        planner_context: &mut PlannerContext,
+    ) -> Result<Expr> {
+        let mut exprs = vec![];
+        entries.into_iter().try_for_each(|entry| {
+            exprs.push(self.sql_expr_to_logical_expr(
+                *entry.key,
+                schema,
+                planner_context,
+            )?);
+            exprs.push(self.sql_expr_to_logical_expr(
+                *entry.value,
+                schema,
+                planner_context,
+            )?);
+            Ok::<_, DataFusionError>(())
+        })?;
```

Comment:
Can we do something like this ? 
```rust
        let mut exprs =  entries.into_iter().flat_map(|entry| {
            let k = self.sql_expr_to_logical_expr(
                *entry.key,
                schema,
                planner_context,
            )?;
            let v = self.sql_expr_to_logical_expr(
                *entry.value,
                schema,
                planner_context,
            )?;
            vec![k,v]
        }).collect<>();;

```
Haven't tried it out just a pseudo code

Reply:
I think it's wrong. What `plan_make_map` accepts is `[k1, v1, k2, v2, ... ]`. The parameters will be arranged to 2 vectors for `map_udf`.
If we change it to `[[k1, k2, ...], [v1, v2, ...]`, it will become an array-to-array map, `MAP {[k1, k2, ...]: [v1, v2, ...]}`.

Reply:
Like i mentioned that was just pseudo code (i should have called it a rough idea). I was try to suggest something like this
```rust
let arr = (1..10).step_by(2).into_iter().flat_map(|e| vec![e,e+1]).collect::<Vec<_>>();
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```
If we can do something like this the expressions iter can be rearrage to `[k1, v1, k2, v2, ... ]`

Reply:
We can also do something like this (again this is a rough idea)

```rust

let exprs: Result<Vec<_>, DataFusionError> = entries
    .into_iter()
    .flat_map(|entry| {
        vec![
            self.sql_expr_to_logical_expr(*entry.key, schema, planner_context),
            self.sql_expr_to_logical_expr(*entry.value, schema, planner_context),
        ]
    })
    .collect()?;
```

Reply:
Got it. I tried to simplify them in e8a3477. It looks better. Thanks.